src/main/java/com/scalar/db/api/Operation.java

```diff
@@ -41,6 +53,22 @@ public Operation(Key partitionKey, Key clusteringKey) {
     return namespace;
   }
 
+  /**
+   * Returns the namespace with the prefix for this operation
+   *
+   * @return an {@code Optional} with the returned namespace with the prefix
+   */
+  @Nonnull
+  public Optional<String> forFullNamespace() {
+    if (namespace.isPresent() && namespacePrefix.isPresent()) {
```

Comment:
I think it should at least warn if this method is called when `namespace` is not present but `namespacePrefix` is present.
What do you think?

Reply:
I've added warning meesages when the `namespace` is not present.